R/get_data.R

```diff
@@ -67,7 +67,7 @@ get_data <- function(x, ...) {
       # remove response for random effects
       if (effects == "random") {
         resp <- find_response(x, combine = FALSE)
-        dat <- dat[setdiff(colnames(dat), resp)]
+        dat <- dat[, setdiff(colnames(dat), resp)]
```

Comment:
I think this doesn't work, as you need the drop argument, while my approach always returned columns:

``` r
iris[setdiff(colnames(iris), "Species")] |> head()
#>   Sepal.Length Sepal.Width Petal.Length Petal.Width
#> 1          5.1         3.5          1.4         0.2
#> 2          4.9         3.0          1.4         0.2
#> 3          4.7         3.2          1.3         0.2
#> 4          4.6         3.1          1.5         0.2
#> 5          5.0         3.6          1.4         0.2
#> 6          5.4         3.9          1.7         0.4
x <- iris[c(1, 5)]
x[setdiff(colnames(x), "Species")] |> head()
#>   Sepal.Length
#> 1          5.1
#> 2          4.9
#> 3          4.7
#> 4          4.6
#> 5          5.0
#> 6          5.4
x[, setdiff(colnames(x), "Species")] |> head()
#> [1] 5.1 4.9 4.7 4.6 5.0 5.4
```

<sup>Created on 2022-11-27 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>